### ISSUE TITLE
Implementation: Deployment of Ambient Light Sensor & fixing sensor issues

### ISSUE THREAD

**Issue by danialhbma** [danialhbma/ITP-SE12-Power-Monitoring#50]
Heading down to RP to solve several issues:
1. Deploy ambient light sensor to track visitor schedule 
2. Test power clamp for literature review
3. Fix m5sticks that went down on rack 1 
4. Check on Rack 3's timer to solve the 10-11pm issue

**Comment by danialhbma** [comment]
![Image](https://github.com/danialhbma/ITP-SE12-Power-Monitoring/assets/92083488/8c583d38-8f27-4445-98fe-32bc9b381a60)

**Comment by danialhbma** [comment]
1. Ambient sensor deployed, readings are sent to InfluxDB. In our tests, we found that the readings will give ~270 lumens when the grow lights are on and the ceiling lights are off, while it gives ~3000 lumens when the ceiling lights are on. We set the timing interval to be 1 minute to take into account short visits that the 30-minute interval readings would not normally give.
2. Power clamp tested with m5stick, could not obtain any readings on the farm as well. Not sure what is the issue with this, would either have to check with prof Fauzi again or note it down as unreliable. Will take note of this test in the final report.
3. Our m5sticks went down on 13/7/23, exactly when the crop scientists replaced the racks with new plants. Our devices went down because the power source was off for some period of time while they replaced the plants, and there was no one on-site to reconfigure them to the right settings (type of reading -> rack number). Could be a potential issue the team can explore to solve.
4. The timer used on rack 3 was not set by the crop scientists to turn on at 10-11 pm. They were also unaware that it was turning on at such a time. Without the ability to observe the lights at 10-11 pm, we concluded that this might be a specific issue with the power monitoring device (tuya) and might be a specific model issue. Will take this into account in our literature review.
5. Another observation made: the portable router used has a limited amount of concurrent users before it cuts off connection to all devices. The DL-7203E is specified to withstand a maximum of 16 connected devices, but when we were doing our tests where we had 7x m5sticks, 6x Tuya devices, 1x laptop & 1x phone connected (a total of 15 devices), the wifi signal was disrupted. Might be a potential limitation factor if the size of the container & number of sensors were to increase.
6. In our initial observation, we thought that the colour of LED strips used was for specific reasons (new/recent plants have purple lights, while older plants have white lights). However, after clarifying with the crop scientists, we found out that they were using the purple lights because they ran out of white lights. The lecturer in charge of the plants prefer to have the white lights on some plants instead, but we could not find a justification for the choice.
